### ISSUE TITLE
Need support for lazy init

### ISSUE THREAD

**Issue by glensc** [moogar0880/PyTrakt#150]
The current design of the module is kind of harsh. Just importing types or any other symbol will make the code to evaluate tokens and execute code.

So I can't even code a flow to initialize app and ask for tokens.

I.e such file header, just importing symbols for typing:

```py
import trakt

trakt.core.CONFIG_PATH = '.pytrakt.json'
from trakt.movies import Movie
```

```
Traceback (most recent call last):
  File "t.py", line 4, in <module>
    from trakt.movies import Movie
  File "/home/glen/.local/share/virtualenvs/test-hmlkD5s0/lib/python3.8/site-packages/trakt/movi
es.py", line 5, in <module>
    from trakt.sync import (Scrobbler, comment, rate, add_to_history,
  File "/home/glen/.local/share/virtualenvs/test-hmlkD5s0/lib/python3.8/site-packages/trakt/sync
.py", line 18, in <module>
    def comment(media, comment_body, spoiler=False, review=False):
  File "/home/glen/.local/share/virtualenvs/test-hmlkD5s0/lib/python3.8/site-packages/trakt/core
.py", line 453, in inner
    _validate_token(args[0])
  File "/home/glen/.local/share/virtualenvs/test-hmlkD5s0/lib/python3.8/site-packages/trakt/core
.py", line 374, in _validate_token
    _refresh_token(s)
  File "/home/glen/.local/share/virtualenvs/test-hmlkD5s0/lib/python3.8/site-packages/trakt/core
.py", line 412, in _refresh_token
    raise s.error_map[response.status_code](response)
trakt.errors.BadRequestException: Bad Request - request couldn't be parsed
```

**Comment by glensc** [comment]
TL;DR, the intermediate need is to import classes like `Movie` without annotations like `@delete` being present in any dependent imports being evaluated. this probably means that the classes need to be moved to new modules that have no dependencies, and the global functions with such annotations are loaded elsewhere.

**Comment by glensc** [comment]
Went with solution in https://github.com/Taxel/PlexTraktSync/pull/314

**Comment by glensc** [comment]
@moogar0880 do you have any ideas on how to refactor this?

the major problem is that core.py can't be loaded without being executed, it gets executed due `@get` and `@post` decorators, which also have `@_boostrapped` and that core.py already instantiates `Core()` and assigns `get`, `post`, `put`. the explanation is "simplification", but it just makes code more difficult to understand:
- https://github.com/moogar0880/PyTrakt/blob/3.2.2/trakt/core.py#L616-L622

and having such decorators will make `load_config`, etc code to be executed, so there's no way to override global constants defined in core.py before code execution.

and import loops, like `trakt/__init__.py` imports [`core.*`](https://github.com/moogar0880/PyTrakt/blob/3.2.2/trakt/__init__.py#L4) and then `trakt.shows` imports `trakt`...

**Comment by glensc** [comment]
the hack I used in this project:
- https://github.com/Taxel/PlexTraktSync/pull/314

works until the user has `~/.pytrakt.json` file present, if the file is present, then load_config will get executed because this clause becomes true:
- https://github.com/moogar0880/PyTrakt/blob/3.2.2/trakt/core.py#L424-L425

load_config is executed every time a method is decorated with _boostrapped:
- https://github.com/moogar0880/PyTrakt/blob/3.2.2/trakt/core.py#L444

**Comment by glensc** [comment]
Something that seems to be rather easily implemented is to move the @_boostrapped decorator:
- https://github.com/moogar0880/PyTrakt/pull/162

so, it's gets executed when the body of `@get`,`@post` is being executed, not at module loading time.
